Gems/Atom/Feature/Common/Code/Source/ProfilingCaptureSystemComponent.cpp

```diff
@@ -590,6 +600,50 @@ namespace AZ
             return captureStarted;
         }
 
+        bool ProfilingCaptureSystemComponent::BeginContinuousCpuProfilingCapture()
+        {
+            if (m_cpuDataSerializationInProgress.load())
+            {
+                AZ_TracePrintf(
+                    "ProfilingSystemCaptureComponent",
+                    "Cannot begin a continuous capture - another serialization is currently in progress\n");
+                return false;
+            }
+            return AZ::RHI::CpuProfiler::Get()->BeginContinuousCapture();
+        }
+
+        bool ProfilingCaptureSystemComponent::EndContinuousCpuProfilingCapture(const AZStd::string& outputFilePath)
+        {
+            if (m_cpuDataSerializationInProgress.load())
+            {
+                AZ_TracePrintf(
+                    "ProfilingSystemCaptureComponent",
+                    "Cannot end a continuous capture - another serialization is currently in progress\n");
+                return false;
+            }
+
+            const bool captureEnded = AZ::RHI::CpuProfiler::Get()->EndContinuousCapture(m_lastContinuousCapture);
+            if (!captureEnded)
+            {
+                AZ_TracePrintf("ProfilingCaptureSystemComponent", "Could not end the continuous capture, is one in progress?\n");
+                return false;
+            }
+
+            // cpuProfilingData could be 1GB+ once saved, so use an IO thread to write it to disk.
```

Comment:
Note: please be super critical of my use of threading here, I haven't used threading primitives before so my use is likely incorrect 😁